pkg/storage/etcd/etcd_watcher.go

```diff
@@ -359,9 +359,6 @@ func (w *etcdWatcher) sendAdd(res *etcd.Response) {
 		return
 	}
 	action := watch.Added
-	if res.Node.ModifiedIndex != res.Node.CreatedIndex {
```

Comment:
I agree with @liggitt that it is baking assumptions into the code. However, since we are going towards etcd v3 pretty soon anyway, I think we should just add comments to explain this in the code. WDYT?